src/worker.cc

```diff
@@ -769,6 +769,34 @@ void from_json(rapidjson::Document& doc, Options& options) {
 
   // costing defaults to none which is only valid for locate
   auto costing_str = rapidjson::get<std::string>(doc, "/costing", "none");
+
+  // auto_shorter is deprecated and will be turned into
+  // shortest=true costing option. maybe remove in v4?
+  if (costing_str == "auto_shorter") {
+    costing_str = "auto";
+    rapidjson::Pointer("/costing").Set(doc, "auto");
```

Comment:
nit: consistent use of `rapidjson` shortcut methods, i.e. `SetValueByPointer`. I'll change that quickly